js/ui/appIconBar.js

```diff
@@ -1011,6 +1064,20 @@ const AppIconBar = new Lang.Class({
         this._panel.closeActiveMenu();
     },
 
+    _activateNthApp: function(index) {
+        if (index == 0) {
+            this._browserButton.activateFirstWindow();
+            return;
+        }
+        this._scrolledIconList.activateNthApp(index - 1);
+    },
+
+    _activateLastApp: function() {
+        // Activate the index of the last button in the scrolled list + 1, to
+        // include the browser button
+        this._activateNthApp(this._scrolledIconList.getNumAppButtons());
```

Comment:
Can do but I think you still need something like `getNumAppButtons` or `hasAppButtons` cause of the browser special casing. If there's no articles in the scrolledIconList `Super + 9` should open the browser, though a bit of a pedantic case.